RepoDb.Core/RepoDb/ClassProperty.cs

```diff
@@ -37,7 +37,7 @@ public class ClassProperty : IEquatable<ClassProperty>
         /// <summary>
         /// Gets the original declaring type (avoiding the interface collision).
         /// </summary>
-        public Type DeclaringType { get; }
+        private readonly Type declaringType;
```

Comment:
Why do it this way? In which context it has been used within the `ClassProperty`?

Reply:
Except for an exception message that uses this attribute, nowhere will it be used internally.
https://github.com/mikependon/RepoDB/blob/4b050185bc8a564ef6ff5e79064c4f4b31f06c4d/RepoDb.Core/RepoDb/Reflection/Compiler/Compiler.cs#L81

The change to an internal property is to force the use of .GetDeclaringType() to avoid confusion caused by the different return values of DeclaringType and GetDeclaringType().